### ISSUE TITLE
bgcolor incompatible when using xleftmargin and/or xrightmargin

### ISSUE THREAD

**Issue by natzei** [gpoore/minted#214]
Hi all. The bug is pretty obvious from the images. Any workaround?

w/o margins:
![schermata del 2018-09-06 16-21-08](https://user-images.githubusercontent.com/4339506/45163792-6daba400-b1f1-11e8-989a-9b70c70132da.png)

w/ margins:
![schermata del 2018-09-06 16-21-38](https://user-images.githubusercontent.com/4339506/45163804-73a18500-b1f1-11e8-8b03-3851f619c206.png)

**Comment by gpoore** [comment]
Can you provide the source for your example?

**Comment by natzei** [comment]
Sure.

```
\documentclass[]{article}

\usepackage[usenames,dvipsnames]{xcolor}
\usepackage{minted}

\setminted{%
    frame=single,
    bgcolor=lightgray
}

%opening
\title{}
\author{}

\begin{document}

\begin{minted}[]{c}
    hello world
\end{minted}


\begin{minted}[xleftmargin=0.1\textwidth,xrightmargin=0.1\textwidth]{c}
hello world
\end{minted}

\end{document}
```

Result:
![schermata del 2018-09-06 17-13-56](https://user-images.githubusercontent.com/4339506/45166893-502e0880-b1f8-11e8-8938-d4b3a129d6df.png)

**Comment by StevenTang1998** [comment]
Does anyone get any solutions?

**Comment by muzimuzhi** [comment]
Try this:
```tex
\documentclass{article}
\usepackage{minted}

\setminted{
    frame=single,
}

\makeatletter
\renewenvironment{minted@snugshade*}[1]{%
  \def\FrameCommand##1{\hskip\@totalleftmargin
    %% old code
    % \colorbox{#1}{##1}%
    %% new code
    \hskip\FV@XLeftMargin
    \colorbox{#1}{\hskip-\FV@XLeftMargin##1\hskip-\FV@XRightMargin}%
    \hskip\FV@XRightMargin
    %% end new code
    \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
  \MakeFramed{\advance\hsize-\width
    \@totalleftmargin\z@ \linewidth\hsize
    \advance\labelsep\fboxsep
    \@setminipage}%
 }{\par\unskip\@minipagefalse\endMakeFramed}
\makeatother

\begin{document}

\begin{minted}[bgcolor=lightgray]{c}
    hello world
\end{minted}

\begin{minted}[xleftmargin=0.1\textwidth,xrightmargin=0.2\textwidth]{c}
hello world
\end{minted}

\begin{minted}[
  bgcolor=lightgray,
  xleftmargin=0.1\textwidth,
  xrightmargin=0.2\textwidth
]{c}
hello world
\end{minted}

\end{document}
```

![image](https://github.com/gpoore/minted/assets/6376638/890021a6-a02b-4144-ac89-ac99f7fb7063)

Note in general `tcolorbox`'s `minted` library provides more powerful hence better control on  decorating `minted` environment output.

**Comment by StevenTang1998** [comment]
Wonderful! Thanks so much for your help!

**Comment by gpoore** [comment]
This is now fixed in `minted` v3.0, which is available as a [beta on GitHub](https://github.com/gpoore/minted/tree/v3-dev) and will have a final release soon. The `minted` v3.0 beta currently requires the [dev version of `fvextra`](https://github.com/gpoore/fvextra/blob/master/fvextra/fvextra.sty). A new `fvextra` release will be on CTAN within the next few days.
